### PR TITLE
feat(driving_environment_analyzer): add new package

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Add `driving_environment_analyzer` tool.

## Related links

Related PR: https://github.com/autowarefoundation/autoware.universe/pull/6496

## Tests performed

Test with ROSBAG.

```
[component_container-1] [INFO 1708999777.768870564] [driving_environment_analyzer]: ======================================
[component_container-1] [INFO 1708999777.768922452] [driving_environment_analyzer]:  data is ready. start ODD analysis... 
[component_container-1] [INFO 1708999777.768933574] [driving_environment_analyzer]: ======================================
[component_container-1] [INFO 1708999777.768967412] [driving_environment_analyzer]: - Length of total lanes : 2357.50 [m]
[component_container-1] [INFO 1708999777.769031174] [driving_environment_analyzer]: - Length of lane that has adjacent lane : 2080.43 [m]
[component_container-1] [INFO 1708999777.769076141] [driving_environment_analyzer]: - Length of lane that has opposite lane : 0.00 [m]
[component_container-1] [INFO 1708999777.769101793] [driving_environment_analyzer]: - Length of lane that has no adjacent lane : 277.07 [m]
[component_container-1] [INFO 1708999777.769225729] [driving_environment_analyzer]: - Min lane width: 3.14 [m] Max lane width: 4.94 [m]
[component_container-1] [INFO 1708999777.769278698] [driving_environment_analyzer]: - Max curvature: 0.007967 [1/m]
[component_container-1] [INFO 1708999777.769293161] [driving_environment_analyzer]: - Min curve radius: 125.52 [m]
[component_container-1] [INFO 1708999777.769336094] [driving_environment_analyzer]: - Min elevation angle: -0.033037 [rad] Max elevation angle: 0.026073 [rad]
[component_container-1] [INFO 1708999777.769403870] [driving_environment_analyzer]: - Min speed limit: 13.89 [m/s] Max speed limit: 16.67 [m/s]
[component_container-1] [INFO 1708999777.769424648] [driving_environment_analyzer]: - Exist traffic light: true
[component_container-1] [INFO 1708999777.769435813] [driving_environment_analyzer]: - Exist intersection: true
[component_container-1] [INFO 1708999777.769620035] [driving_environment_analyzer]: - Exist crosswalk: true
[component_container-1] [INFO 1708999777.769634980] [driving_environment_analyzer]: ======================================
[component_container-1] [INFO 1708999777.769642769] [driving_environment_analyzer]:  complete ODD analysis. shutdown. 
[component_container-1] [INFO 1708999777.769650034] [driving_environment_analyzer]: ======================================
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: driving_environment_analyzer/include/driving_environment_analyzer/node.hpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using geometry_msgs::msg::AccelWithCovarianceStamped;
+using nav_msgs::msg::Odometry;
+
+class DrivingEnvironmentAnalyzer : public rclcpp::Node
+{
+public:
+  explicit DrivingEnvironmentAnalyzer(const rclcpp::NodeOptions & node_options);
+
+private:
+  bool isEnoughData(const bool use_map_in_bag);
```
Comment (takayuki5168) [inline driving_environment_analyzer/include/driving_environment_analyzer/node.hpp:52]: nits: hasEnoughData or isDataReady

File: driving_environment_analyzer/src/node.cpp [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+    return false;
+  }
+
+  return true;
+}
+
+void DrivingEnvironmentAnalyzer::onMap(const HADMapBin::ConstSharedPtr msg)
+{
+  route_handler_.setMap(*msg);
+  has_map_data_ = true;
```
Comment (takayuki5168) [inline driving_environment_analyzer/src/node.cpp:337]: nits:
route_handler_ and has_map_data_ can be converted to one variable by making route_handler_ std::optional?

**Review by takayuki5168 — APPROVED** [review]
LGTM!

File: driving_environment_analyzer/include/driving_environment_analyzer/node.hpp (hunk 1 above)
Comment (satoshi-ota) [reply driving_environment_analyzer/include/driving_environment_analyzer/node.hpp]: Fixed in https://github.com/autowarefoundation/autoware_tools/pull/13/commits/c52479950020059dafd26ce94b9bddd9f3ca03d7.

**Pull request merged**